src/packet.rs

```diff
@@ -48,33 +50,47 @@ enum MspParserState {
 	Header1,
 	Header2,
 	Direction,
+	FlagV2,
 	DataLength,
+	DataLengthV2,
 	Command,
+	CommandV2,
 	Data,
-	Crc
+	DataV2,
+	Crc,
+}
+
+#[derive(Copy, Clone, PartialEq, Debug)]
+enum MSPVersion {
+		V1,
+		V2,
 }
 
 #[derive(Debug)]
 /// Parser that can find packets from a raw byte stream
 pub struct MspParser {
 	state: MspParserState,
+	packet_version: MSPVersion,
 	packet_direction: MspPacketDirection,
-	packet_cmd: u8,
-	packet_data_length_remaining: usize,
+	packet_cmd: u16,
+	packet_data_length_remaining: u16,
```

Comment:
why `u16` here? There is no way how the length of an buffer/slice/array is going to be indexed by something else then `usize`, for the better and the worse.

Reply:
no particular reason, reverting back to use usize